src/Transport/Addressing/Partitioning/Strategies/RoundRobinNamespacePartitioning.cs

```diff
@@ -3,30 +3,24 @@ namespace NServiceBus
     using System;
     using System.Collections.Generic;
     using System.Configuration;
-    using System.Linq;
+    using AzureServiceBus.Addressing.Partitioning;
     using Settings;
     using Transport.AzureServiceBus;
 
     public class RoundRobinNamespacePartitioning : INamespacePartitioningStrategy
     {
-        CircularBuffer<NamespaceInfo> namespaces;
+        CircularBuffer<RuntimeNamespaceInfo> namespaces;
 
         public RoundRobinNamespacePartitioning(ReadOnlySettings settings)
         {
-            NamespaceConfigurations namespaces;
-            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Namespaces, out namespaces))
-            {
-                throw new ConfigurationErrorsException($"The '{nameof(RoundRobinNamespacePartitioning)}' strategy requires more than one namespace, please use {nameof(AzureServiceBusTransportExtensions.NamespacePartitioning)}().{nameof(AzureServiceBusNamespacePartitioningSettings.AddNamespace)}() to register multiple namespaces");
```

Comment:
We're losing the error message which would explain why strategy is not configured properly.